diff --git a/CRM/Member/Page/Tab.php b/CRM/Member/Page/Tab.php
index 7ab06f1a95..72e831b578 100644
--- a/CRM/Member/Page/Tab.php
+++ b/CRM/Member/Page/Tab.php
@@ -148,7 +148,7 @@ class CRM_Member_Page_Tab extends CRM_Core_Page {
             'id' => $dao->id,
             'cid' => $this->_contactId,
           ],
-          ts('Renew') . '...',
+          'more',
           FALSE,
           'membership.tab.row',
           'Membership',
